src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -285,6 +296,7 @@ private async Task ForceAuthenticationAsync<TIOAdapter>(TIOAdapter adapter, bool
                 if (reAuthenticationData == null)
                 {
                     _nestedAuth = 0;
+                    _handshakeBuffer.Dispose();
```

Comment:
Why don't we need to dispose this if reAuthenticationData != null?  Also, should we set _handshakeBuffer to default after it's been disposed?

Reply:
no, we should. I missed that part.